src/python/gudhi/simplex_tree.pyx

```diff
@@ -184,6 +184,15 @@ cdef class SimplexTree:
         simplices are already present with a higher filtration value, their
         filtration value is lowered.
 
+        .. note::
+
+            Inserting a simplex with filtration value `math.nan` does not
+            modify the filtration value of any simplex already present.
+            However, it produces an invalid filtration, which needs to be fixed
+            with :meth:`make_filtration_non_decreasing` or
+            :meth:`assign_filtration`. Inserting simplices with real filtration
+            value may or may not update pre-existing NaN.
```

Comment:
It seems to me that inserting a simplices will never update pre-existing NaN.
```python
st.insert([0,1], math.nan)  # return True, all is NaN
st.insert([0,1], 1.)  # return False
list(st.get_filtration())  # returns:  [([0, 1], nan), ([0], nan), ([1], nan)]
st.insert([1,2], 1.)  # return True
list(st.get_filtration())  # returns: [([0, 1], nan), ([0], nan), ([1], nan), ([2], 1.0), ([1, 2], 1.0)]
```

Reply:
Yes, but I don't think that's desirable behavior, so I don't want to document a promise for that behavior if we are likely to break it later.